worker/docker_client.py

```diff
@@ -171,27 +169,79 @@ def test(self):
             if version_info['ApiVersion'] < '1.17':
```

Comment:
careful, lexical comparisons have bitten me before (and that occurred precisely when I was comparing docker versions)